common/lib/xmodule/xmodule/js/src/video/01_initialize.js

```diff
@@ -532,12 +532,9 @@ function (VideoPlayer, VideoStorage, i18n) {
 
             this.youtubeXhr
                 .always(function (json, status) {
-                    var err = $.isPlainObject(json.error) ||
-                                (
-                                    status !== 'success' &&
-                                    status !== 'notmodified'
-                                );
-                    if (err) {
+                    // It will work for both if statusCode is 200 or 410.
+                    var didSucceed = (json.error && json.error.code === 410) || status === 'success';
```

Comment:
I think we should also add  `|| status === 'notmodified'` as well to match the existing checks.